### DISCUSSION TITLE
Few elements to improve the user experience of table component

### DISCUSSION THREAD

**Discussion by ElodieOS** [PrestaShop/PrestaShop#32975] [Ideas]
### The problem

Today, table component is a central BO tool that is no longer at all optimised for the needs of merchants. We have display problems that prevent rapid analysis of the data, responsive problems that prevent multi-device use and a general lack of efficiency due to the simplicity of the functions offered.

### The solution

There are three things we can do to quickly improve the experience: 
- **Horizontal scrolling**: if we can apply horizontal scrolling for all devices, we can display more information and in a better way. We're not limited by the content and we can better manage the width of the columns. To maintain control over scrolling, we can define maximum column sizes depending on the content. For example, a small size with a maximum width of 112px for checkboxes, IDs, images, toggles or actions. A medium size, which will be the default size, with a maximum width of 240px. A large size for the most important content, such as the description, message, name or email, with a maximum width of 480px. 

- **Fix the column**: if we have horizontal scrolling, we need to fix the columns to allow quick action by the merchant. For mobile screens, we don't fix the columns so as not to interfere with reading, for screens smaller than 1024px, only the action column will be fixed and for all other screens, we can fix the action and checkboxes columns. 

- If we can, we must change the design of the filter. If we use the same mechanism as the combination settings in the product page, we can display more content and have better harmony with the rest of the BO.

![image](https://github.com/PrestaShop/PrestaShop/assets/113457393/61c3c8f2-df2f-47ed-8979-ead4dc4e242e)
![image](https://github.com/PrestaShop/PrestaShop/assets/113457393/bc6cac41-27be-44b5-91ab-6b16cecf027d)

Wdyt? 

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Love it. Maybe we should also ask the PrestaShop Edition team, it would be a great addition to the software, and maybe we could work together on this.

**Reply by MatShir** [reply]
I already tried, the mission failed!

**Comment by leemyongpakvn** [comment]
I see we discussed this problem quite a lot in the past, especially at [2020 - Responsive in the back office](https://github.com/PrestaShop/PrestaShop/issues/22358) and [2021 -  Implement mobile version of tables](https://github.com/PrestaShop/PrestaShop/discussions/32280). There are 2 proposals for grid responsive in these 2 discussions: horizontal scrolling and card view. Horizontal scrolling seems an old technique while Card view is more popular. 
In fact, Legacy HelperList already solved Table responsive with Card view in `/admin-dev/index.php?controller=AdminCartRules`. We will replace Lagacy HelperList by Symfony Grid in the future, so we can not rely on HelperList. Symfony Grid in turn, does not support responsive well as we can see in `/admin-dev/index.php/_profiler/`, so we can not rely on it either.
Till now, I made two PR to solve Table responsive issue, [one with Bootstrap grid](https://github.com/PrestaShop/blockreassurance/pull/432) and [other with Stacking responsive table](https://github.com/PrestaShop/pagesnotfound/pull/24)

**Reply by MatShir** [reply]
True for mobile this debate needs to find a end, agree. IMO here we are more focus on desktop / tablet screen view.

**Comment by Hlavtox** [comment]
@ElodieOS We already have a finished design for tables and their filters here - https://github.com/PrestaShop/PrestaShop/discussions/32280#discussioncomment-5686893

**Reply by MatShir** [reply]
Here we are more focalised on desktop / Tablet screen
